src/converters.ts

```diff
@@ -5,18 +5,100 @@ import {
   ConversionResponse,
   ConversionValue,
   Converter,
-  IConverter
+  IConverter,
+  StateConverterOptionsWithContext
 } from "./converter";
 import { Controlled, controlled } from "./controlled";
 import { identity } from "./utils";
 
 const NUMBER_REGEX = new RegExp("^-?(0|[1-9]\\d*)(\\.\\d*)?$");
 const INTEGER_REGEX = new RegExp("^-?(0|[1-9]\\d*)$");
 
+function getLastElement(
+  splitRaw: string[],
+  options: StateConverterOptionsWithContext
+) {
+  if (options.decimalSeparator == null) {
+    return splitRaw[splitRaw.length - 1];
+  }
+  return splitRaw[splitRaw.length - 1].split(options.decimalSeparator)[0];
+}
+
+function convertDecimalSeparator(
+  raw: string,
+  options: StateConverterOptionsWithContext
+) {
+  if (options.decimalSeparator == null) {
+    return raw;
+  }
+  return raw.replace(options.decimalSeparator, ".");
+}
+
+function renderDecimalSeparator(
+  value: string,
+  options: StateConverterOptionsWithContext
+) {
+  if (options.decimalSeparator == null) {
+    return value;
+  }
+  return value.replace(".", options.decimalSeparator);
+}
+
+function convertThousandSeparators(
+  raw: string,
+  options: StateConverterOptionsWithContext
+) {
+  if (options.thousandSeparator == null) {
+    return raw;
+  }
+  const splitRaw = raw.split(options.thousandSeparator);
+  const firstElement = splitRaw[0];
+  const lastElement = getLastElement(splitRaw, options);
+  //value before the first thousand separator has to be of length 1, 2 or 3
+  if (firstElement.length < 1 || firstElement.length > 3) {
+    return raw;
+  }
+  //all other separated values should have length 3
+  for (let i = 1; i < splitRaw.length - 1; i++) {
+    if (splitRaw[i].length !== 3) {
+      return raw;
+    }
+  }
+  if (lastElement.length !== 3) {
+    return raw;
+  }
+  //turn split string back into full string without thousand separators
+  return splitRaw.join("");
+}
+
+function renderThousandSeparators(
+  value: string,
+  options: StateConverterOptionsWithContext
+) {
+  if (options.thousandSeparator == null || !options.renderThousands) {
+    return value;
+  }
+  return value.replace(/\B(?=(\d{3})+(?!\d))/g, options.thousandSeparator);
+}
+
+function processSeparators(
```

Comment:
Call this 'convertSeparators'. Introduce the equivalent logic 'renderSeparators' as you repeat yourself a bit below.